src/handlers/miner_hbbft_handler.erl

```diff
@@ -21,22 +21,43 @@
         {
          n :: non_neg_integer(),
          f :: non_neg_integer(),
-         id :: non_neg_integer(),
+         id :: non_neg_integer(), % TODO Remove? Set at init, but never accessed.
          hbbft :: hbbft:hbbft_data(),
          sk :: tc_key_share:tc_key_share() | tpke_privkey:privkey() | binary(),
-         seq = 0,
-         deferred = [],
-         signatures = [],
+         seq = 0,        % TODO Remove? Looks unused.
+         deferred = [],  % TODO Remove? Looks unused.
+         sigs_valid     = [] :: sigs(),
+         sigs_invalid   = [] :: sigs(),
+         sigs_unchecked = [] :: sigs(),
          signatures_required = 0,
          sig_phase = unsent :: unsent | sig | gossip | done,
          artifact :: undefined | binary(),
          members :: [libp2p_crypto:pubkey_bin()],
-         chain :: undefined | blockchain:blockchain(),
+         chain :: undefined | blockchain:blockchain(), % TODO Remove? Set at init, but never accessed.
```

Comment:
Remove if not used, yes

Reply:
The couple of times chain is used (in handling `skip` and `next_round`), it is grabbed from `blockchain_worker:blockchain()`, which blocks. Seems like it would be better to start using the chain from state to replace those calls instead of removing. What do you think? @Vagabond @evanmcc

Reply:
yes, that is also fine